encoding/Encoding.java

```diff
@@ -476,6 +478,31 @@ public static ValueType<?> of(TypeQLArg.ValueType typeQLValueType) {
             else throw TypeDBException.of(UNRECOGNISED_VALUE);
         }
 
+        public static <T, U> int compare(ValueType<T> firstType, T firstValue, ValueType<U> secondType, U secondValue) {
```

Comment:
our new generic, centralised method for comparing across value types. In the future, we can extend this to handle more types.

In addition, should `ValueType` grow, we can split it into actual sub-classes, eg. `Encoding.ValueType.Long/Double/String/Boolean/DateTime` so that these kind of switching logic and the comparables & assignables maps above can be absorbed.